mmdet3d/core/bbox/coders/anchor_free_bbox_coder.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+import torch
+
+from mmdet.core.bbox.builder import BBOX_CODERS
+from .partial_bin_based_bbox_coder import PartialBinBasedBBoxCoder
+
+
+@BBOX_CODERS.register_module()
+class AnchorFreeBBoxCoder(PartialBinBasedBBoxCoder):
```

Comment:
consider to merge AnchorFreeBBoxCoder into PartialBinBasedBBoxCoder instead of inheriting it. use `if self.with_...` to control operation in encode, decode and split_pred.